src/components/network-area-diagram-viewer/diagram-utils.ts

```diff
@@ -64,6 +65,224 @@ const EdgeTypeMapping: { [key: string]: EdgeType } = {
 const TEXT_BOX_WIDTH_DEFAULT = 200.0;
 const TEXT_BOX_HEIGHT_DEFAULT = 100.0;
 
+export function getBendableFrom(element: SVGElement): SVGElement | undefined {
+    if (isBendable(element)) {
+        return element;
+    } else if (element.parentElement) {
+        return getBendableFrom(element.parentNode as SVGElement);
+    }
+}
+
+export function isBendable(element: SVGElement): boolean {
+    return hasId(element) && element.parentNode != null && idIs(element.parentNode as SVGElement);
+}
+
+function idIs(element: SVGElement): boolean {
+    return element.id == 'lines-points';
+}
+
+export function getLinePointId(edgeId: string | undefined, index: number): string {
+    return edgeId + '-point-' + index;
+}
+
+export function getEdgeId(
+    linePointIndexMap: Map<SVGGElement, { edgeId: string; index: number }>,
+    linePoint: SVGGElement
+): string | undefined {
+    return linePointIndexMap.get(linePoint)?.edgeId;
+}
+
+export function getLinePointMapKey(edgeId: string, index: number): string {
```

Comment:
This is used as key for the map `linePointByEdgeIndexMap`, but you should use instead the edgeId as key and an array as value: the index will give you the element.

Concatenation of strings for indexing/ids should in most cases be avoided